app/controllers/graphql_devise/concerns/additional_controller_methods.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module GraphqlDevise
+  module Concerns
+    module AdditionalControllerMethods
+      extend ActiveSupport::Concern
+
+      included do
+        attr_accessor :client_id, :token, :resource
+      end
+
+      def gql_devise_context(models)
+        {
+          current_resource: authenticate_model(models),
+          controller:       self
+        }
+      end
+
+      def authenticate_model(models)
+        Array(models).each do |model|
+          set_resource_by_token(model)
+          return @resource if @resource.present?
+        end
+
+        nil
+      end
+
+      def resource_class(resource = nil)
+        # Return the resource class instead of looking for a Devise mapping if resource is already a resource class
```

Comment:
I think this comment should mention `find_by` is meant to use duck typing to indentify AR and Mongo ID models.